### PR TITLE
Redact api and app keys from errors

### LINKED ISSUE #76: POST errors leak credentials
Discovered via: https://github.com/hashicorp/terraform/issues/11107

When a POST request comes into the DataDog API via: https://github.com/hashicorp/terraform/blob/master/builtin/providers/datadog/resource_datadog_monitor.go#L227, the `go-datadog-api` calls `*http.Client.Do()`. 

Due to the fact that the API Key and Application Key are provided in the URI, and the URI is included inside the returned Error; this leads to leaking credentials on the CLI where the error is returned. 

cc: @keymon who reported the issue to Terraform.

### PR DESCRIPTION
**Pull request by nyanshak**
Fix #76.

I don't know if this is the right way to fix this but I took a stab at it.

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -41,9 +42,25 @@ func (client *Client) uriForAPI(api string) (string, error) {
 }
 
 // doJsonRequest is the simplest type of request: a method on a URI that returns
-// some JSON result which we unmarshal into the passed interface.
+// some JSON result which we unmarshal into the passed interface. It wraps
+// doJsonRequestUnsafe to redact api and application keys from errors.
 func (client *Client) doJsonRequest(method, api string,
 	reqbody, out interface{}) error {
+	err := client.doJsonRequestUnsafe(method, api, reqbody, out)
```
Comment (ojongerius) [inline request.go]: A nit picking alternative would be:

```go
   if err := client.doJsonRequestUnsafe(method, api, reqbody, out) ; err != nil {
        errString := strings.Replace(err.Error(), client.apiKey, "redacted", -1)
        errString = strings.Replace(errString, client.appKey, "redacted", -1)
        return fmt.Errorf(errString)
    }
```

File: request.go [hunk 2]
```diff
@@ -41,9 +42,25 @@ func (client *Client) uriForAPI(api string) (string, error) {
...
+		errString = strings.Replace(errString, client.apiKey, "redacted", -1)
+		errString = strings.Replace(errString, client.appKey, "redacted", -1)
+		return fmt.Errorf("%s", errString)
+	}
+	return nil
+}
+
+// doJsonRequestUnsafe is the simplest type of request: a method on a URI that returns
+// some JSON result which we unmarshal into the passed interface.
+func (client *Client) doJsonRequestUnsafe(method, api string,
```
Comment (ojongerius) [inline request.go]: Would `doJsonRequestUnredacted` be a more informative name?

File: request.go (hunk 1 above)
Comment (nyanshak) [reply request.go]: Will update this ^

**Comment by ojongerius** [comment]
@nyanshak do you have time to add a test to make sure this issue does not regress? Making this change would be a  good time to add it.

**Comment by nyanshak** [comment]
Rebased off master to resolve conflicts, also added the tests you mentioned @ojongerius. What do you think? :)

**Comment by zorkian** [comment]
Awesome, thanks for adding the tests! LGTM!

**Pull request merged**